src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,70 @@
+package gift.controller;
+
+import gift.dto.ProductRequestDto;
+import gift.dto.ProductResponseDto;
+import gift.entity.Product;
+import gift.service.ProductService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    // 전체 조회
+    @GetMapping
+    public List<ProductResponseDto> productList() {
+        return productService.findAll()
+                .stream()
+                .map(ProductResponseDto::new)
+                .collect(Collectors.toList());
```

Comment:
stream 을 활용하셨군요? 👍💯 

자바 16에서부터는 Stream에 `toList()` 메서드가 추가됐는데요. 
이 메서드를 활용해본다면 지금과 같이 스트림을 리스트로 변경할 경우 이전보다 코드를 더 간결하게 활용할 수 있을 것 같아요. 😃

```suggestion
        return productService.findAll()
                .stream()
                .map(ProductResponseDto::new)
                .toList();
```